lib/web_ui/lib/src/engine/window.dart

```diff
@@ -774,3 +814,28 @@ class ViewConstraints implements ui.ViewConstraints {
     return 'ViewConstraints($width, $height)';
   }
 }
+
+// Computes the "min" value for a constraint that takes into account user configuration
+// and the actual available size.
+//
+// Returns the configured userValue, unless it's null (not passed) in which it returns
+// the actual physicalSize.
+double _computeMinConstraintValue(double? desired, double available) {
+  assert(desired == null || desired >= 0, 'Minimum constraint cannot be less than 0');
+  return desired ?? available;
+}
+
+// Computes the "max" value for a constraint that takes into account user configuration
+// and the available size.
+//
+// Returns the configured userValue unless:
+//  * It is null, in which case it returns the physicalSize
+//  * It is `-1`, in which case it returns "infinity" / unconstrained.
+double _computeMaxConstraintValue(double? desired, double available) {
+  assert(desired == null || desired >= -1, 'Maximum constraint must be greater than 0 (or -1 for unconstrained)');
+  return switch (desired) {
+    null => available,
+    -1 => double.infinity,
```

Comment:
I'm curious why we are using `-1` to represent infinity instead of JavaScript's `Infinity`? Is it because JS-interop doesn't handle the `Infinity` value well?

Reply:
Good point, I don't know how JS interop handles `Infinity`, but we need to ensure we can handle the value in our code.

Reply:
We now support `Infinity` instead of `-1`. It has simplified the code quite a bunch!